compiler/rustc_middle/src/ty/query/mod.rs

```diff
@@ -188,25 +188,13 @@ pub fn force_from_dep_node<'tcx>(tcx: TyCtxt<'tcx>, dep_node: &DepNode) -> bool
                 }
 
                 $(DepKind::$name => {
-                    debug_assert!(<$K as DepNodeParams<TyCtxt<'_>>>::can_reconstruct_query_key());
```

Comment:
Shouldn't this be kept?

Reply:
The assert has been moved inside `foce_query`.